services/movie_session.py

```diff
@@ -0,0 +1,41 @@
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+from datetime import datetime
+
+
+def create_movie_session(movie_show_time: datetime, movie_id: int, cinema_hall_id: int) -> MovieSession:
+    movie = Movie.objects.get(id=movie_id)
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=movie,
+        cinema_hall=cinema_hall
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    if session_date:
+        date_obj = datetime.strptime(session_date, "%Y-%m-%d").date()
+        return MovieSession.objects.filter(show_time__date=date_obj)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int, show_time: str = None, movie_id: int = None, cinema_hall_id: int = None) -> MovieSession:
+    session = MovieSession.objects.get(id=session_id)
+    if show_time:
+        session.show_time = show_time
```

Comment:
The `show_time` parameter is expected to be a string, but it should be converted to a `datetime` object before being assigned to `session.show_time`. Consider using `datetime.strptime(show_time, "%Y-%m-%d %H:%M:%S")` to convert it properly.